source/extensions/filters/network/redis_proxy/proxy_filter.cc

```diff
@@ -57,6 +58,35 @@ void ProxyFilter::initializeReadFilterCallbacks(Network::ReadFilterCallbacks& ca
 }
 
 void ProxyFilter::onRespValue(Common::Redis::RespValuePtr&& value) {
+  std::cout << "ProxyFilter::onRespValue" << std::endl;
+  
+  // Fault injection
+  std::string command = "get"; // TODO: Put RedisCommandStats::getCommandFromRequest(...) into utils or resp something
+
+  absl::optional<std::pair<Common::Redis::FaultType, std::chrono::milliseconds>> fault = config_->fault_manager_.get_fault_for_command(command);
+  if (fault.has_value()) {
+
+    if (fault.value().first == Common::Redis::FaultType::Delay) {
+      std::cout << "\t" << "--- FAULT INJECTION - [DELAY] ---" << std::endl;
+      return;
+    } else if (fault.value().first == Common::Redis::FaultType::Error) {
+      std::cout << "\t" << "--- FAULT INJECTION - [ERROR] ---" << std::endl;
+      // TODO: Downstream stats
+      Common::Redis::RespValuePtr response{new Common::Redis::RespValue()};
+      response->type(Common::Redis::RespType::Error);
+      response->asString() = "Fault Injection: Abort";
+      connection_allowed_ = true;
+      encoder_->encode(*response, encoder_buffer_);
+      if (encoder_buffer_.length() > 0) {
+        callbacks_->connection().write(encoder_buffer_, false);
```

Comment:
directly writing to the downstream is not the correct behavior, 1) it might be out of order, 2)we might have to drain/close the downstream connection.

Reply:
Done